cmd/agent/entrypoint.go

```diff
@@ -352,6 +352,22 @@ func (ep *Entrypoint) TriggerReload() bool {
 	return true
 }
 
+// Triggers a reload of the config on each tick of the ticker until the context
+// completes.
+func (ep *Entrypoint) PollConfig(ctx context.Context, t *time.Ticker) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
Generally we only return `ctx.Err()` if the canceled context means something failed. In this case, it's expected for the function to poll forever until the context is canceled, so this should be `return nil`.